### ISSUE TITLE
mergeAdjacentRules wrongly merges ::-webkit-input-placeholder and ::-moz-placeholder for some selectors

### ISSUE THREAD

**Issue by ngyikp** [jakubpawlowicz/clean-css#891]
clean-css@4.0.6

Just `.z::-webkit-input-placeholder` is fine, but adding `[data-x="y"]` in front of it causes a merge with `::-moz-placeholder` which shouldn't happen.

Input:
```css
[data-x="y"] .z::-webkit-input-placeholder {color:#000;}
[data-x="y"] .z::-moz-placeholder {color:#000;}
```

$ ./node_modules/.bin/cleancss style.css -O2
```css
[data-x="y"] .z::-moz-placeholder,[data-x="y"] .z::-webkit-input-placeholder{color:#000}
```

$ ./node_modules/.bin/cleancss style.css -O2 "mergeAdjacentRules:off"
```css
[data-x="y"] .z::-webkit-input-placeholder{color:#000}[data-x="y"] .z::-moz-placeholder{color:#000}
```

**Comment by jakubpawlowicz** [comment]
This should not happen.

**Comment by jakubpawlowicz** [comment]
Fixed in 4.0.7.
